plugins/ign-docker-env.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018 David Rosa
+# Licensed under the Apache License, Version 2.0
+
+
+# gzdev docker-env --ignition citadel
+# gzdev docker-env --ignition dome --linux-distro ubuntu:focal
+# gzdev docker-env --ignition dome --docker-args '-v /foo:/foo'
+
+"""
+Usage:
+    gzdev ign-docker-env IGN_RELEASE
+                [--linux-distro <linux-distro>]
+                [--docker-args DOCKER_ARGS]
+    gzdev ign-docker-env -h | --help
+    gzdev ign-docker-env --version
+
+Options:
+    -h --help                   Show this screen
+    --version                   Show gzdev's version
+    --linux-distro=linux-distro Linux distibution to use in docker env
+    --docker-args DOCKER_ARGS   Extra arguments to pass to docker
+"""
+
+from docopt import docopt
+from subprocess import check_call
+from sys import stderr
+import os
+
+
+ROCKER_CMD = ['rocker', '--x11', '--user']
+
+
+def detect_nvidia():
+    return [x for x in os.listdir('/dev/') if x.startswith('nvidia')] != []
+
+
+def _check_call(cmd):
+    print('')
+    print("Invoking '%s'" % ' '.join(cmd))
+    print('')
+
+    try:
+        check_call(cmd)
+    except Exception as e:
+        print(str(e))
+
+
+def error(msg):
+    print(f"\n {msg} \n", file=stderr)
+    exit(-1)
+
+
+# TODO: use a yaml file to get this information
+def default_distro_by_ignition(ignition_release):
+    if ignition_release == 'dome':
+        return 'ubuntu:focal'
+    elif ignition_release == 'citadel':
+        return 'ubuntu:bionic'
+    else:
+        error(f"Unknown ignition release {ignition_release}")
+
+
+def build_rocker_command(igniton_release, linux_distro, docker_args):
+    _, linux_distro_release = linux_distro.split(':')
+    cmd = ROCKER_CMD + ['--nvidia'] if detect_nvidia() else ROCKER_CMD
+    cmd += ['--ignition', f"{igniton_release}:{linux_distro_release}"]
+    cmd += docker_args if docker_args else []
+    cmd += ['--']
```

Comment:
I'm don't think that having a `--docker-args` argument will work. If you take a look at rocker's default extensions, they don't support arbitrary docker arguments:
```
optional arguments:
  -h, --help            show this help message and exit
  --noexecute           Deprecated (default: False)
  --nocache
  --pull
  -v, --version         show program's version number and exit
  --dev-helpers         add development tools emacs and byobu to your environment (default: None)
  --devices [DEVICES [DEVICES ...]]
                        Mount devices into the container. (default: None)
  --env NAME[=VALUE] [NAME[=VALUE] ...], -e NAME[=VALUE] [NAME[=VALUE] ...]
                        set environment variables (default: [])
  --env-file ENV_FILE   set environment variable via env-file (default: None)
  --git                 Use the global Git settings from the host (/etc/gitconfig and ~/.gitconfig)
                        (default: None)
  --home                mount the users home directory (default: None)
  --network {host,bridge,none}
                        What network configuration to use. (default: None)
  --nvidia              Enable nvidia (default: None)
  --pulse               mount pulse audio devices (default: None)
  --ssh                 Forward SSH agent into the container (default: None)
  --user                mount the current user's id and run as that user (default: None)
  --x11                 Enable x11 (default: None)
  --mode {interactive,non-interactive,dry-run}
                        Choose mode of operation for rocker (default: interactive)
  --extension-blacklist [EXTENSION_BLACKLIST [EXTENSION_BLACKLIST ...]]
                        Prevent any of these extensions from being loaded. (default: [])
```

However, we can at least incorporate the `--volume` option that is offered by `ign-rocker`:
```
  --vol /LOCAL/PATH:/CONTAINER/PATH [/LOCAL/PATH:/CONTAINER/PATH ...]
                        Mount files on your machine into the container via Docker volumes. This must be
                        the last argument given before the image name. After all volumes are given,
                        terminate this argument with '--' before the image name. (default: None)
```

I suggest that we replace this optional `--docker-args` argument with `--volume`, which can serve as a wrapper for the `--volume` argument from `ign-rocker`. That way, users can have a way to load source code into containers via volumes, which is the only thing they really need with this setup anyways.

Reply:
I was trying to avoid the exact duplication of the rocker interface. As an alternative, what about to correct the name and use `rocker-args` this way we can use it for injecting 1..N arguments from the rocker interface. See the test in da4cbc9

Reply:
I like the idea of using `rocker-args` more than `docker-args`. The one tricky thing is that the `--volume` argument has to end with `--` since multiple volumes can be passed in (separated by white space), and the `--volume` argument must be the last argument in the underlying `rocker` command since it ends with `--`. So, if we use something like `rocker-args`, we will need to update the parsing logic of this a bit to place `--vol` last (if it exists) and make sure it's followed by `--` before the image name.

However, I'm still wondering if it would make sense to just have a `--vol` argument that users can call directly. The reason why I argue for this is because this setup is mainly targeted towards users who need to load source code into the container (either for building Ignition from source or for using a project with an Ignition dependency). So, if volumes are going to be used most of the time, I feel like having to type in `--rocker-args '--vol ...'` is inconvenient - I'd rather just type in `--vol ...`. Also, another reason why I would rather have a direct `--vol` argument is because when typing things inside of quotes (which you'd have to do for `--rocker-args`), you don't get tab completion for things like file/directory paths, but using `--vol` would allow for users to have tab completion when specifying their local volume path. I think this is a very important feature to have in case users have long/complex local paths they are trying to mount, and since the whole purpose of this wrapper is to simplify the usage of `ign-rocker`, I think that using something like `--rocker-args '...'` doesn't make volume mounting any easier - **_so, I propose this:_** we can have both a `--rocker-args` argument and a `--vol` argument, and if users want to mount volumes, it should be done through `--vol`, while everything else can be done through `--rocker-args` (this would also make parsing easier - all we'd have to do is add in everything listed in `--rocker-args` first, and then add in whatever was specified in the `--vol` arg, followed by `--`, and then end the command with the image name and default `/bin/bash` command).

Reply:
Sounds reasonable to me. :+1: